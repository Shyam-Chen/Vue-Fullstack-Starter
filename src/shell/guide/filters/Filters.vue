<template>
  <v-container class="shell-container" fluid>
    <v-layout column>
      <div class="headline mb-1">Filters</div>

      <v-layout column class="mt-1 mb-1">
        <div class="title mb-2">Reverse String/Array</div>

        <div>{{ 'Vanilla Angular React Vue' | reverse }}</div>

        <template v-for="item in reverse(list)">
          <div :key="item">{{ item }}</div>
        </template>
      </v-layout>

      <v-layout column class="mt-1 mb-1">
        <div class="title mb-2">Truncate String</div>

        <div>{{ 'Approachable' | truncate }}</div>
        <div>{{ message | truncate }}</div>
        <div>{{ message | truncate(50) }}</div>
      </v-layout>

      <v-layout column class="mt-1 mb-1">
        <div class="title mb-2">Convert Time</div>

        <div>{{ 44230 | convertSeconds }}</div>
        <div>{{ 4230 | convertSeconds }}</div>
        <div>{{ 777 | convertSeconds }}</div>
        <div>{{ 390 | convertSeconds }}</div>
        <div>{{ 12 | convertSeconds }}</div>
        <div>{{ 3 | convertSeconds }}</div>
      </v-layout>

      <v-layout column class="mt-1 mb-1">
        <div class="title mb-2">Time Ago</div>

        <div>{{ Date.now() | timeSince }}</div>
        <div>{{ 1529823939463 | timeSince }}</div>
        <div>{{ '2013-01-04' | timeSince }}</div>
      </v-layout>
    </v-layout>
  </v-container>
</template>

<script>
// @flow

import reverse from './reverse';
import truncate from './truncate';
import convertSeconds from './convert-seconds';
import timeSince from './time-since';

export default {
  filters: {
    reverse,
    truncate,
    convertSeconds,
    timeSince,
  },
  data() {
    return {
      list: ['Vanilla', 'Angular', 'React', 'Vue'],
      message:
        'Already know HTML, CSS and JavaScript? Read the guide and start building things in no time!',
    };
  },
  methods: {
    reverse,
  },
};
</script>

<style scoped>
.layout:not(:first-child) {
  padding: 1rem;
}
</style>
