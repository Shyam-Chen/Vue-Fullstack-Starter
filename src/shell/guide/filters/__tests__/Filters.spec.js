import { createLocalVue, shallowMount } from '@vue/test-utils';
import mockdate from 'mockdate';

import Filters from '../Filters';

const localVue = createLocalVue();

describe('Filters', () => {
  let [wrapper] = [];

  beforeEach(() => {
    wrapper = shallowMount(Filters, {
      localVue,
      stubs: ['router-link', 'router-view'],
      mocks: {
        Date: mockdate.set('2020-01-10'),
      },
    });
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
