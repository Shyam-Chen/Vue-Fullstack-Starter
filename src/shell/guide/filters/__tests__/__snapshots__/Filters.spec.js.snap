// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should render an initial component 1`] = `
<v-container fluid="" class="shell-container">
  <v-layout column="">
    <div class="headline mb-1">Filters</div>
    <v-layout column="" class="mt-1 mb-1">
      <div class="title mb-2">Reverse String/Array</div>
      <div>euV tcaeR ralugnA allinaV</div>
      <div>Vue</div>
      <div>React</div>
      <div>Angular</div>
      <div>Vanilla</div>
    </v-layout>
    <v-layout column="" class="mt-1 mb-1">
      <div class="title mb-2">Truncate String</div>
      <div>Approachable</div>
      <div>Already know HT...</div>
      <div>Already know HTML, CSS and JavaScript? Read the gu...</div>
    </v-layout>
    <v-layout column="" class="mt-1 mb-1">
      <div class="title mb-2">Convert Time</div>
      <div>12:17:10</div>
      <div>01:10:30</div>
      <div>12:57</div>
      <div>06:30</div>
      <div>00:12</div>
      <div>00:03</div>
    </v-layout>
    <v-layout column="" class="mt-1 mb-1">
      <div class="title mb-2">Time Ago</div>
      <div>a few seconds ago</div>
      <div>2 years ago</div>
      <div>7 years ago</div>
    </v-layout>
  </v-layout>
</v-container>
`;
